core/static/user/js/family_graph.js

```diff
@@ -0,0 +1,267 @@
+    console.error(
+      "ctxmenu isn't loaded, context menu won't be available on graphs",
+    );
+    return cy;
+  }
+  cy.cxtmenu({
+    selector: "node",
+
+    commands: [
+      {
+        content: '<i class="fa fa-external-link fa-2x"></i>',
+        select: function (el) {
+          window.open(el.data().profile_url, "_blank").focus();
+        },
+      },
+
+      {
+        content: '<span class="fa fa-mouse-pointer fa-2x"></span>',
+        select: function (el) {
+          on_node_tap(el);
+        },
+      },
+
+      {
+        content: '<i class="fa fa-eraser fa-2x"></i>',
+        select: function (el) {
+          reset_graph();
+        },
+      },
+    ],
+  });
+
+  return cy;
+}
+
+document.addEventListener("alpine:init", () => {
+  /*
+    This needs some constants to be set before the document has been loaded
+
+    api_url:     base url for fetching the tree as a string
+    active_user: id of the user to fetch the tree from
+    depth_min:   minimum tree depth for godfathers and godchildren as an int
+    depth_max:   maximum tree depth for godfathers and godchildren as an int
+  */
+  const default_depth = 2;
+
+  if (
+    typeof api_url === "undefined" ||
+    typeof active_user === "undefined" ||
+    typeof depth_min === "undefined" ||
+    typeof depth_max === "undefined"
+  ) {
+    console.error("Some constants are not set before using the family_graph script, please look at the documentation");
+    return;
+  }
+
+  function get_initial_depth(prop) {
+    let value = parseInt(initialUrlParams.get(prop));
+    if (isNaN(value) || value < depth_min || value > depth_max) {
+      return default_depth;
+    }
+    return value;
+  }
+
+  Alpine.data("graph", () => ({
+    loading: false,
+    godfathers_depth: get_initial_depth("godfathers_depth"),
+    godchildren_depth: get_initial_depth("godchildren_depth"),
+    reverse: initialUrlParams.get("reverse")?.toLowerCase?.() === 'true',
+    graph: undefined,
+    graph_data: {},
+
+    async init() {
+      let delayed_fetch = Alpine.debounce(async () => {
+        this.fetch_graph_data();
+      }, 100);
+      ["godfathers_depth", "godchildren_depth"].forEach((param) => {
+        this.$watch(param, async (value) => {
+          if (value < depth_min || value > depth_max) {
+            return;
+          }
+          update_query_string(param, value, History.REPLACE);
+          delayed_fetch();
+        });
+      });
+      this.$watch("reverse", async (value) => {
+        update_query_string("reverse", value, History.REPLACE);
+        this.reverse_graph();
+      });
+      this.$watch("graph_data", async () => {
+        await this.generate_graph();
+        if (this.reverse) {
+          await this.reverse_graph();
+        }
+      });
+      this.fetch_graph_data();
+    },
+
+    async screenshot() {
+      const link = document.createElement("a");
+      link.href = this.graph.jpg();
+      link.download = "output.jpg";
```

Comment:
```suggestion
      link.download = "family_tree.jpg";
```
Voire même `gettext("family_tree.jpg")`, soyons fous.